### PR TITLE
Grouping by original order id in Link analysis

### PR DESCRIPTION
**Pull request by irinatarn**
Before:
image

After:
image

### CODE REVIEW

File: frontend/src/screens/review-console/item-details/parts/link-analysis/items-list/items-list.tsx [hunk 1]
```diff
@@ -326,19 +329,58 @@ export class ItemsList extends Component<ItemsListComponentProps, never> {
             this.selection.setAllSelected(false);
             return this.renderLoader();
         }
+        const sortedData = data
+            .slice()
+            .sort((prev, next) => {
+                const prevX = isMrItem(prev)
+                    ? prev.item?.purchase?.merchantLocalDate
+                    : prev.merchantLocalDate;
```
Comment (haixiehx) [inline frontend/src/screens/review-console/item-details/parts/link-analysis/items-list/items-list.tsx:337]: Not `prev.item.purchase.merchantLocalDate` at line 337 and 340, that's what you checked nullable on?

File: frontend/src/screens/review-console/item-details/parts/link-analysis/items-list/items-list.tsx [hunk 2]
```diff
@@ -326,19 +329,58 @@ export class ItemsList extends Component<ItemsListComponentProps, never> {
...
+                const nextX = isMrItem(next)
+                    ? next.item?.purchase?.merchantLocalDate
+                    : next.merchantLocalDate;
+                const prevDate = formatDateStringToJSDate(prevX);
```
Comment (haixiehx) [inline frontend/src/screens/review-console/item-details/parts/link-analysis/items-list/items-list.tsx:341]: If `prevX` is not a valid date or null, I assume `formatDateStringToJSDate` will take care and return null, is that true?

File: frontend/src/screens/review-console/item-details/parts/link-analysis/items-list/items-list.tsx [hunk 3]
```diff
@@ -326,19 +329,58 @@ export class ItemsList extends Component<ItemsListComponentProps, never> {
...
+        Object.keys(groupsData).forEach((id: string) => {
+            const currentGroup = groupsData[id];
+            currentGroup.forEach((i: LinkAnalysisItem) => groupedItems.push(i));
+
+            groups.push({
+                key: `group${id}`,
+                name: `Original order ID: ${id}`,
+                startIndex: groupIndex,
+                count: currentGroup.length,
+                level: 1
```
Comment (haixiehx) [inline frontend/src/screens/review-console/item-details/parts/link-analysis/items-list/items-list.tsx:366]: just curious, what is level about?

**Review by haixiehx — APPROVED** [review]

File: frontend/src/screens/review-console/item-details/parts/link-analysis/items-list/items-list.tsx (hunk 1 above)
Comment (irinatarn) [reply frontend/src/screens/review-console/item-details/parts/link-analysis/items-list/items-list.tsx:337]: yes.

File: frontend/src/screens/review-console/item-details/parts/link-analysis/items-list/items-list.tsx (hunk 3 above)
Comment (irinatarn) [reply frontend/src/screens/review-console/item-details/parts/link-analysis/items-list/items-list.tsx:366]: you can have tree. We need only one level tree.
see https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/grouped

**Pull request merged**